app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -381,13 +381,7 @@ def get_content_view_environments(label = nil, organization = nil)
           where("#{Organization.table_name}.id = ?", organization.id)
       environments = environments.where("#{Katello::ContentViewEnvironment.table_name}.label = ?", label) if label
 
-      environments.delete_if do |env|
-        if env.content_view.default
-          !env.environment.readable?
-        else
-          !env.content_view.readable?
-        end
-      end
+      environments = environments.reject { |env| (env.content_view.default && !env.environment.readable?) || !env.content_view.readable? }
```

Comment:
@cpeters Unfortunately, delete_if was deprecated for active record relations in [Rails 4.1](http://edgeguides.rubyonrails.org/4_1_release_notes.html#active-record-notable-changes) 

> Relation no longer has mutator methods like #map! and #delete_if.